docs/api.md

```diff
@@ -1,50 +1,98 @@
+.. automodule:: pytket.extensions.qiskit.backends.ibm_utils
+.. automodule:: pytket.extensions.qiskit.backends.ibm
+.. autoclass:: pytket.extensions.qiskit.backends.ibm.IBMQBackend
     :show-inheritance:
-    :members:
+
+    .. automethod:: __init__
+    .. automethod:: available_devices
+    .. automethod:: cancel
+    .. automethod:: circuit_status
+    .. automethod:: default_compilation_pass
+    .. automethod:: default_compilation_pass_offline
+    .. automethod:: get_compiled_circuit
+    .. automethod:: get_compiled_circuits
+    .. automethod:: get_result
+    .. automethod:: process_circuits
+    .. automethod:: rebase_pass
+    .. automethod:: rebase_pass_offline
+    .. automethod:: squash_pass_offline
+    .. autoproperty:: backend_info
+    .. autoproperty:: required_predicates
+
+.. autoexception:: pytket.extensions.qiskit.backends.ibm.NoIBMQCredentialsError
 ```
 
 ```{eval-rst}
-.. autoclass:: pytket.extensions.qiskit.IBMQEmulatorBackend
-    :special-members: __init__
+.. automodule:: pytket.extensions.qiskit.backends.ibmq_emulator
+.. autoclass:: pytket.extensions.qiskit.backends.ibmq_emulator.IBMQEmulatorBackend
     :show-inheritance:
-    :members:
+
+    .. automethod:: __init__
+    .. automethod:: cancel
+    .. automethod:: circuit_status
+    .. automethod:: default_compilation_pass
+    .. automethod:: get_result
+    .. automethod:: process_circuits
+    .. automethod:: rebase_pass
+    .. autoproperty:: backend_info
+    .. autoproperty:: required_predicates
 ```
 
 ```{eval-rst}
-.. autoclass:: pytket.extensions.qiskit.AerBackend
+.. automodule:: pytket.extensions.qiskit.backends.aer
+.. autoclass:: pytket.extensions.qiskit.backends.aer.AerBackend
     :special-members: __init__
     :inherited-members:
     :members:
 ```
 
 ```{eval-rst}
-.. autoclass:: pytket.extensions.qiskit.AerStateBackend
+.. autoclass:: pytket.extensions.qiskit.backends.aer.AerStateBackend
     :special-members: __init__
     :inherited-members:
     :members:
 ```
 
 ```{eval-rst}
-.. autoclass:: pytket.extensions.qiskit.AerUnitaryBackend
+.. autoclass:: pytket.extensions.qiskit.backends.aer.AerUnitaryBackend
     :special-members: __init__
     :inherited-members:
     :members:
 ```
 
 ```{eval-rst}
-.. autoclass:: pytket.extensions.qiskit.AerDensityMatrixBackend
+.. autoclass:: pytket.extensions.qiskit.backends.aer.AerDensityMatrixBackend
     :special-members: __init__
     :inherited-members:
     :members:
 ```
 
 ```{eval-rst}
-.. automodule:: pytket.extensions.qiskit
-    :members: qiskit_to_tk, tk_to_qiskit, process_characterisation
+.. autofunction:: pytket.extensions.qiskit.backends.aer.qiskit_aer_backend
+.. autoclass:: pytket.extensions.qiskit.backends.aer.NoiseModelCharacterisation
+    :members:
+```
+
+```{eval-rst}
+.. automodule:: pytket.extensions.qiskit.qiskit_convert
+
+    .. autofunction:: qiskit_to_tk
+    .. autofunction:: tk_to_qiskit
+    .. autofunction:: process_characterisation
```

Comment:
I think `param_to_qiskit` and `param_to_tk` can probably be underscored. They are just helpers for the circuit conversion code.